### ISSUE TITLE
Switch statement does not flag non-comparable cases with non-overlapping literal and non-literal union types

### ISSUE THREAD

**Issue by ericbf** [microsoft/TypeScript#38686]
**TypeScript Version:**  3.0.0-dev.201xxxxx

**Search Terms:** 
Switch, non-comparable types, false negative

**Code**

```ts
declare let message: number | "hello"

switch (message) {
    case "hi":
        break
}
```

**Expected behavior:**
Type '"hi"' is not comparable to type 'number | "hello"'.
(2678)

**Actual behavior:**
No error.

**Playground Link:** https://www.typescriptlang.org/play/?ssl=1&ssc=31&pln=1&pc=22#code/CYUwxgNghgTiAEEQBd4FsQGdNQOYgC54A7AVzQCMQZ4AfeAIgAsQIIB7BgKC8wHcAlsjBN4ACgzY8IAJTwA3l3jL4YKJgTMBDAkpX6KcKAGsuAXyA

**Related Issues:** None that I found

**Comment by hoangpham95** [comment]
Can I take a stab at this issue?

**Comment by ericbf** [comment]
Please!

**Comment by ericbf** [comment]
This is still an issue on at least 4.8.4:

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEEQBd4FsQGdNQOYgC54A7AVzQCMQZ4AfeAIgAsQIIB7BgKC8wHcAlsjBN4ACgzY8IAJTwA3lwCQYKJgTMBDAsqUU4UANZcAvkA

**Comment by ericbf** [comment]
I opened a PR that fixes this issue. May need some assistance with ascertaining any unexpected side effects.
